master/master.cfg

```diff
@@ -13,35 +13,37 @@
 import os
 import sys
 
+from datetime import timedelta
+
 from buildbot.schedulers.basic import SingleBranchScheduler
 from buildbot.schedulers.timed import Nightly
 from buildbot.schedulers.forcesched import (ForceScheduler, FixedParameter,
                                             StringParameter, UserNameParameter)
+from buildbot.plugins import reporters, util
 from buildbot.process import factory
-from buildbot.status import html, mail, words
 from buildbot.steps.shell import ShellCommand, Configure, Compile
 from buildbot.steps.transfer import DirectoryUpload
 from buildbot import locks
 from buildbot.steps.source.git import Git
+from buildbot.www.auth import RemoteUserAuth
 
 sys.path.append(os.path.dirname(__file__))   # noqa: E402
 
 from custom.steps import Test, Clean, Install, LockInstall, Uninstall
 
 from utils import changecommentlink
-from xmlrpc import XMLRPCServer
 # sensitive and/or instance-specific values
 try:
-    from local import (SLAVES, AUTHZ, SLAVE_PORT, WEB_PORT, DEBUG_PASSWORD,
+    from local import (WORKERS, AUTH, WORKER_PORT, WEB_PORT, DEBUG_PASSWORD,
                        BUILDBOT_URL, PRODUCTION, IRC_NICK, IRC_CHANNEL,
                        GIT_URL, GITHUB_SECRET)
 except ImportError:
-    from buildbot.plugins import buildslave
-    from buildbot.status.web.authz import Authz
+    from buildbot.plugins import worker
+    from buildbot.www.auth import RemoteUserAuth
     PRODUCTION = False
-    SLAVES = [buildslave.BuildSlave('test-slave', 'badsecret')]
-    AUTHZ = Authz()
-    SLAVE_PORT = 9021
+    WORKERS = [worker.Worker('test-worker', 'badsecret')]
+    AUTH = RemoteUserAuth()
```

Comment:
Let's go with `NoAuth` as the default here for now.  We'll figure out something reasonable to put in `local.py` when we deploy this, and then figure out a nice way to factor it so that only the secrets are in `local.py`.

Reply:
Fixed